src/pools/gyroEPool/gyroEMath/gyroSignedFixedPoint.ts

```diff
@@ -0,0 +1,206 @@
+import { BigNumber } from '@ethersproject/bignumber';
+import { WeiPerEther as ONE } from '@ethersproject/constants';
+import {
+    SQRT_1E_NEG_1,
+    SQRT_1E_NEG_3,
+    SQRT_1E_NEG_5,
+    SQRT_1E_NEG_7,
+    SQRT_1E_NEG_9,
+    SQRT_1E_NEG_11,
+    SQRT_1E_NEG_13,
+    SQRT_1E_NEG_15,
+    SQRT_1E_NEG_17,
+    ONE_XP,
+} from './constants';
+
+/////////
+/// ARITHMETIC HELPERS
+/////////
+
+export function mulUp(a: BigNumber, b: BigNumber) {
```

Comment:
I think a lot of this is repeated functions that already exist in other helpers - could you consolidate them please?